server/src/main/java/io/crate/role/TransportAlterRoleAction.java

```diff
@@ -115,10 +128,16 @@ static boolean alterRole(Metadata.Builder mdBuilder, String roleName, @Nullable
         boolean exists = false;
         var role = newMetadata.roles().get(roleName);
         if (role != null) {
-            if (role.isUser() == false && secureHash != null) {
-                throw new UnsupportedFeatureException("Setting a password to a ROLE is not allowed");
+            if (role.isUser() == false) {
+                if (secureHash != null || resetPassword) {
+                    throw new UnsupportedFeatureException("Setting a password to a ROLE is not allowed");
+                }
+                if (jwtProperties != null || resetJwtProperties) {
+                    throw new UnsupportedFeatureException("Setting JWT properties to a ROLE is not allowed");
+                }
             }
```

Comment:
On the other hand, this logic could move into the the `role.with` or its constructor to prevent creating invalid role instances.

Reply:
I the comment above I moved out reset logic, ie `with` takes `finalized` values.
Moving this logic to `with` would again make it reset flags aware.

In ctor also not good as we need to make ctor reset flags aware then.

Reply:
Please check 2 last commits. I don't like this change as error message became a bit vague, see
https://github.com/crate/crate/pull/15646/commits/ef7b7e2744df12fa8daf113e4f2aad154fd2541e.

We cannot check which of pwd/jwt is not null to tune the error message because NULL value can be also invalid for Role (reset).

Reply:
yeah I agree, that makes things worse. Better to remove it again and leave it as it was initially